src/mlpack/tests/serialization_test.cpp

```diff
@@ -1690,4 +1693,97 @@ BOOST_AUTO_TEST_CASE(HoeffdingTreeTest)
   }
 }
 
+/**
+ * Build a Binary RBM, then save it and make sure the parameters of the
+ * all the RBM are equal.
+ */
+
+BOOST_AUTO_TEST_CASE(BinaryRBMTest)
+{
+  arma::mat data;
+  size_t hiddenLayerSize = 5;
+  data.randu(3, 100);
+
+  GaussianInitialization gaussian(0, 0.1);
+  RBM<GaussianInitialization> Rbm(data, gaussian, data.n_rows, hiddenLayerSize,
+      1, 1, 1, 2, 8, 1, true);
+  RBM<GaussianInitialization> RbmXml(data, gaussian, data.n_rows,
+      hiddenLayerSize, 1, 1, 1, 2, 8, 1, true);
+  RBM<GaussianInitialization> RbmText(data, gaussian, data.n_rows,
+      hiddenLayerSize, 1, 1, 1, 2, 8, 1, true);
+  RBM<GaussianInitialization> RbmBinary(data, gaussian, data.n_rows,
+      hiddenLayerSize, 1, 1, 1, 2, 8, 1, true);
+  Rbm.Reset();
+
+  SerializeObjectAll(Rbm, RbmXml, RbmText, RbmBinary);
+  CheckMatrices(Rbm.Parameters(), RbmXml.Parameters(), RbmText.Parameters(),
+      RbmBinary.Parameters());
+  CheckMatrices(Rbm.VisibleBias(), RbmXml.VisibleBias());
+  CheckMatrices(Rbm.VisibleBias(), RbmText.VisibleBias());
+  CheckMatrices(Rbm.VisibleBias(), RbmBinary.VisibleBias());
+
+  CheckMatrices(Rbm.HiddenBias(), RbmXml.HiddenBias());
+  CheckMatrices(Rbm.HiddenBias(), RbmText.HiddenBias());
+  CheckMatrices(Rbm.HiddenBias(), RbmBinary.HiddenBias());
+
+  CheckMatrices(Rbm.Weight(), RbmXml.Weight());
+  CheckMatrices(Rbm.Weight(), RbmText.Weight());
+  CheckMatrices(Rbm.Weight(), RbmBinary.Weight());
+}
+
+/**
+ * Build a ssRBM, then save it and make sure the parameters of the
+ * all the RBM are equal.
+ */
+
+BOOST_AUTO_TEST_CASE(ssRBMTest)
+{
+  arma::mat data;
+  size_t hiddenLayerSize = 5;
+  data.randu(3, 100);
+  double slabPenalty = 1;
+  double radius, tempRadius;
+  for (size_t i = 0; i < data.n_cols; i++)
+  {
+    tempRadius = arma::norm(data.col(i));
+    if (radius < tempRadius)
```

Comment:
We should initialize radius with ``arma::norm(data.col(0));`` and start with ``i = 1`` in the for loop, to avoid unexpected behavior. Let me know what you think.